src/core/src/op/util/scatter_nd_base.cpp

```diff
@@ -35,10 +35,18 @@ void ov::op::util::ScatterNDBase::validate_and_infer_types() {
     element::Type updates_et = get_input_element_type(UPDATES);
 
     NODE_VALIDATION_CHECK(this,
-                          indices_et == element::i32 || indices_et == element::i64,
+                          indices_et == element::i32 || indices_et == element::i64 || indices_et == element::dynamic,
```

Comment:
Or maybe use compatible?
```suggestion
                          indices_et.compatible(element::i32) || indices_et.compatible(element::i64),
```